app/controllers/users/password_compromised_controller.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module Users
+  class PasswordCompromisedController < ApplicationController
+    before_action :confirm_two_factor_authenticated
+    before_action :verify_feature_toggle_on?
+
+    def show
+      session.delete(:redirect_to_password_compromised)
+      @after_sign_in_path = after_sign_in_path_for(current_user)
+      analytics.user_password_compromised_visited
+    end
+
+    def verify_feature_toggle_on?
```

Comment:
The trailing `?` suffix indicates a [predicate method returning a boolean](https://rubystyle.guide/#bool-methods-qmark), which this is not. I'd suggest dropping.

```suggestion
    def verify_feature_toggle_on
```